### DISCUSSION TITLE
Rebar optimization for Building Construction...

### DISCUSSION THREAD

**Discussion by hakanornek** [google/or-tools#3531] [Linear Solver questions] [answered]
Hello,
I am very newbe to Or-Tools. So I didn't know which method to use for my problem.
I am a civil engineer and we need to use different lengths of rebar in our reinforced concrete structures
The fabrication production length of these rebars arrives at the construction site as 12 meters. Our rebar rods of different lengths required
I want to cut it from bars with a fabricated length of 12 meters with minimum casualties. What kind of solver should I use for this 1-dimensional optimization?
 and if you have C# sample code that can help me with this, I would be very grateful if you could share it.

**Comment by ktnr** [comment, marked as answer]
You are looking for the Cutting Stock Problem. See [this repo](https://github.com/emadehsan/csp) and [this or-tools example](https://github.com/google/or-tools/blob/stable/examples/python/arc_flow_cutting_stock_sat.py).

**Reply by hakanornek** [reply]
Thanks for your helps.
